lesson-4/README.md

```diff
@@ -107,9 +107,96 @@ NAME         TYPE       CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
+NAME                         READY     STATUS    RESTARTS   AGE
+prometheus-647f4b785-lcdzc   1/1       Running   0          1d
+```
+
+Open port-forwaring to Prometheus pod, port 9090
+```bash
+kubectl -n monitoring port-forward prometheus-647f4b785-lcdzc 9090
+```
+
+Open [127.0.0.1:9090/graph](http://127.0.0.1:9090/graph) in web browser
+
+Let's show graph for container memory usage:
+
+1. Click on Graph
+1. Write expression: container_memory_usage_bytes
+1. Press execute - it can take some time if you have many containers running in a cluster
+
+However, query language is much more powerful. Try these expressions:
+* [`container_memory_usage_bytes{namespace="monitoring"}`](http://localhost:9090/graph?g0.range_input=1h&g0.expr=container_memory_usage_bytes%7Bnamespace%3D%22monitoring%22%7D&g0.tab=0)
+* [`kubernetes_build_info`](http://localhost:9090/graph?g0.range_input=1h&g0.expr=kubernetes_build_info&g0.tab=0)
+* [`go_goroutines{kubernetes_namespace="monitoring"}`](http://localhost:9090/graph?g0.range_input=1h&g0.expr=go_goroutines%7Bkubernetes_namespace%3D%22monitoring%22%7D&g0.tab=0)
+* [`predict_linear(node_filesystem_free_bytes[10m], 30*24*3600)`](http://localhost:9090/graph?g0.range_input=1h&g0.expr=predict_linear(node_filesystem_free_bytes%5B10m%5D%2C%2030*24*3600)&g0.tab=0)
+
+### Grafana
+
+Grafana is a tool for visualization and it can load data from many different backends, including Prometheus. Grafana dashboards can be create manually, auto-generated or imported from [Grafana.com Dashboards](https://grafana.com/dashboards). The goal of this part is to configure Prometheus datasource in grafana and explore some dashboard.
+
+First we need to start Grafana pod in monitoring namespace.
+```bash
+kubectl apply -f grafana-deployment.yaml
+```
+
+and check that grafana container is running:
+```bash
+kubectl -n monitoring get po
+NAME                         READY     STATUS    RESTARTS   AGE
+grafana-646f55c587-xlp2n     1/1       Running   0          4m
+```
+
+Open connection to grafana and don't forget to replace `grafana-646f55c587-xlp2n` with name of your grafana pod.
+```bash
+kubectl -n monitoring port-forward grafana-646f55c587-xlp2n 3000
+```
+
+Open [127.0.0.1:3000/](http://127.0.0.1:3000) in web browser
+
+
+We need to first configure data source which will enable grafana to read metrics from Prometheus:
+
+1. Login with username `admin` and password `admin`
+1. You will be asked to change password but it can be skipped
+1. Click `Add data source` and set 
+  * Name: prgcont
+  * Type:  Prometheus
+  * URL: http://prometheus:9090
+* Click "Save & Test"
+
+Next step is to add some dashboards which can be created manually but we will just import predefined dashboard:
+
+1. Move mouse over plus sign on top left side of page
+1. Click Import
+1. Write 6879 to field with title "Grafana.com Dashboard"
+1. Click Load
+1. Select Prometheus datasource prgcont
+1. Click import
+1. Discover provided graphs
```

Comment:
maybe it will be nicer with screenshots?

Reply:
I'l create screenshot for adding prometheus datasource but these steps are really easy. I'll try to add some screenshots